### ISSUE TITLE
Why divide 0.07

### ISSUE THREAD

**Issue by vvthai10** [zqhang/AnomalyCLIP#15]
I don't understand why you divide right here
`
text_probs = image_features.unsqueeze(1) @ text_features.permute(0, 2, 1)
            text_probs = text_probs[:, 0, ...]/0.07
            text_probs = text_probs.squeeze()
`
and another here
`
logit_scale = self.logit_scale.exp()  # nn.Parameter(torch.ones([]) * np.log(1 / 0.07))
        logits_per_image = logit_scale * image_features @ text_features.t()
        logits_per_text = logits_per_image.t()
`

And in another paper , they use multi with 100, example
`
for layer in range(len(det_patch_tokens)):
    det_patch_tokens[layer] = det_patch_tokens[layer] / det_patch_tokens[layer].norm(dim=-1, keepdim=True)
    anomaly_map = (100.0 * det_patch_tokens[layer] @ text_features)
    anomaly_map = torch.softmax(anomaly_map, dim=-1)[:, :, 1]
    anomaly_score = torch.mean(anomaly_map, dim=-1)
    det_loss += loss_bce(anomaly_score, image_label)
`

**Comment by zqhang** [comment]
It is the temperature ratio of the original CLIP.

**Comment by vvthai10** [comment]
But i see in https://github.com/mlfoundations/open_clip#usage, they use 100, you think it make result different?

**Comment by zqhang** [comment]
i dont think it will affect much



---- Replied Message ----
| From | ***@***.***> |
| Date | 05/24/2024 14:30 |
| To | ***@***.***> |
| Cc | Qihang ***@***.***>***@***.***> |
| Subject | Re: [zqhang/AnomalyCLIP] Why divide 0.07 (Issue #15) |

But i see in https://github.com/mlfoundations/open_clip#usage, they use 100, you think it make result different?

**Comment by zqhang** [comment]
you can have a try

**Comment by vvthai10** [comment]
I am trying but i don't have make high ACC, I just high in first epoch and after it reduce and I don't know why :(

**Comment by vvthai10** [comment]
I train with dataset visa,
I train 11 epoch, but loss and image_loss is 3.7960, 0.5325. I feel
something wrong. I setup by your setting. Can you share me about loss and
image_loss when u train?

**Comment by akya-one** [comment]
> I am trying but i don't have make high ACC, I just high in first epoch and after it reduce and I don't know why :(

Hello, I am also working on this. How did you add the Linear layer? Can you give me some reference?
